saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -308,3 +312,245 @@ def perform_mutation(cls, _root, info, **data):
         fulfillment.refresh_from_db(fields=["status"])
         order.refresh_from_db(fields=["status"])
         return FulfillmentCancel(fulfillment=fulfillment, order=order)
+
+
+class OrderRefundLineInput(graphene.InputObjectType):
+    order_line_id = graphene.ID(
+        description="The ID of the order line to refund.", name="orderLineId"
+    )
+    quantity = graphene.Int(
+        description="The number of items to be refunded.", required=True,
+    )
+
+
+class OrderRefundFulfillmentLineInput(graphene.InputObjectType):
+    fulfillment_line_id = graphene.ID(
+        description="The ID of the fulfillment line to refund.",
+        name="fulfillmentLineId",
+    )
+    quantity = graphene.Int(
+        description="The number of items to be refunded.", required=True,
+    )
+
+
+class OrderRefundProductsInput(graphene.InputObjectType):
+    order_lines = graphene.List(
+        graphene.NonNull(OrderRefundLineInput),
+        description="List of unfulfilled lines to refund.",
+    )
+    fulfillment_lines = graphene.List(
+        graphene.NonNull(OrderRefundFulfillmentLineInput),
+        description="List of fulfilled lines to refund.",
+    )
+    amount_to_refund = PositiveDecimal(
+        required=False,
+        description=(
+            "The total amount of refund when the value is provided manually. The "
+            "amountToRefund and includeShippingCosts are mutually exclusive and "
+            "attempts to call mutation with both options will raise an exception."
+        ),
+    )
+    include_shipping_costs = graphene.Boolean(
+        description=(
+            "If true, Saleor will refund shipping costs. The amountToRefund and "
+            "includeShippingCosts are mutually exclusive and attempts to call mutation "
+            "with both options will raise an exception."
+        ),
+        default_value=False,
+    )
+
+
+class FulfillmentRefundProducts(BaseMutation):
+    fulfillment = graphene.Field(Fulfillment, description="A refunded fulfillment.")
+    order = graphene.Field(Order, description="Order which fulfillment was refunded.")
+
+    class Arguments:
+        order_id = graphene.ID(
+            description="ID of the order to be fulfilled.", name="order", required=True
+        )
+        input = OrderRefundProductsInput(
+            required=True,
+            description="Fields required to create an refund fulfillment.",
+        )
+
+    class Meta:
+        description = "Refund products."
+        permissions = (OrderPermissions.MANAGE_ORDERS,)
+        error_type_class = OrderError
+        error_type_field = "order_errors"
+
+    @classmethod
+    def clean_input(cls, info, order_id, input):
+        amount_to_refund = input.get("amount_to_refund")
+        include_shipping_costs = input.get("include_shipping_costs")
```

Comment:
`input["include_shipping_costs"]`? that var has a default so it will always be there.